libs/hdf-converters/src/msft-secure-score-mapper.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  SecureScore,
+  ControlScore,
+  SecureScoreControlProfile
+} from '@microsoft/microsoft-graph-types';
+import {ExecJSON} from 'inspecjs';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+
+export class MsftSecureScoreMapper extends BaseConverter {
+  withRaw: boolean;
+  profiles: SecureScoreControlProfile[];
+
+  private getProfiles(controlName: string): SecureScoreControlProfile[] {
+    return this.profiles.filter((profile) => profile.id === controlName);
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score Scan',
+        title: {
+          transformer: (data: SecureScore) =>
+            `Azure Secure Score report - TenantID: ${data.azureTenantId}`
+        },
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (data: ControlScore) =>
+                `${data.controlCategory}:${data.controlName}`
+            },
+            title: {
+              transformer: (data: ControlScore) => {
+                const titles = this.getProfiles(data.controlName || '')
+                  .filter((profile) => profile.title !== undefined)
+                  .map((profile) => profile.title);
+
+                if (titles.length > 0) {
+                  return titles.join('... ');
+                } else {
+                  return `${data.controlCategory || ''}:${data.controlName || ''}`;
+                }
+              }
+            },
+            desc: {path: 'description'},
+            impact: {
```

Comment:
would profile.deprecated have an impact?  https://github.com/microsoftgraph/msgraph-typescript-typings/blob/main/microsoft-graph.d.ts#L8349-L8350 I am unsure if it being deprecated means that we should set the impact to 0 or if we leave it be.  this'd be a @ejaronne question.

Reply:
1) profile.deprecated is always TRUE in the data that we've seen thus far.
2) `// Flag to indicate if a control is depreciated.`
     `deprecated?: NullableOption<boolean>;`
3) I suspect that if a control is deprecated, it would no longer be presented under the `LIST controlScore` endpoint, rendering this discussion moot.

This might be good to revisit after we first see a deprecated control.

Reply:
We'll worry about this resulting in false positives when someone complains I guess.